core/src/main/java/org/kohsuke/stapler/lang/Klass.java

```diff
@@ -56,7 +57,13 @@ public Class toJavaClass() {
     }
 
     public List<FieldRef> getDeclaredFields() {
-        return navigator.getDeclaredFields(clazz);
+        try {
+            return navigator.getDeclaredFields(clazz);
+        } catch (AbstractMethodError err) {
+            // A plugin uses obsolete version of Stapler-dependent library (e.g. JRuby), which does not offer the method (JENKINS-39414)
+            // TODO: what to do with Logging? The error must be VERY visible, but it will totally pollute system logs
```

Comment:
Both options do not work since we **may** invoke this method dozens and dozens of times during serving a single page. I would rather trigger administrative monitor in Jenkins, but it requires something (e.g. publicly accessible static field with a list of failed navigators). A bit hackish...